lib/utils.js

```diff
@@ -221,3 +221,46 @@ exports.replaceAll = function replaceAll(line, oldValue, newValue) {
     }
     return line;
 };
+
+// Initally based on http://philzimmermann.com/docs/human-oriented-base-32-encoding.txt
+
+const BASE32_ALPHABET = "13456789EGKMPTWYabcdefhjnoqrsvxz";
+/***
+ *  A Base32 encoding implementation valid for our needs
+ * @param {Buffer} buffer
+ * @returns {string}
+ */
+exports.base32Encode = function base32Encode(buffer) {
+    let output = "";
+    // This can grow up to 12 bits
+    let digest = 0;
+    // How many bits are we actually using from digest
+    let bits = 0;
+
+    function encodeNewWord() {
+        // Get first 5 bits
+        const word = digest & 0b11111;
+        const character = BASE32_ALPHABET[word];
+        output += character;
+        bits -= 5;
+        // Shift out the newly processed word
+        digest = (digest >>> 5);
+    }
+
+    for (const byte of buffer) {
+        // Append current byte to digest
+        digest = ((byte) * Math.pow(2, bits)) | digest;
```

Comment:
For my future self: `digest = (byte << bits) | digest`

Reply:
Why can't we use the `(byte << bits)`? What am I forgetting about javascript here haha

Reply:
No reason not to! I even left a note to myself an hour ago to remind me tomorrow to switch it!